### ISSUE TITLE
[Object Detection] ‘faster_rcnn_resnet50_coco’ model is errored during inference, after trt optimization: 'Incompatible shapes: [17328] vs. [16150]'

### ISSUE THREAD

**Issue by oscarriddle** [tensorflow/tensorrt#62]
GPU: 1080Ti
docker: nvcr.io/nvidia/tensorflow   19.03-py3           e8ec6af2fb6f

My test.py:
```

from tftrt.examples.object_detection import download_model
                                                                                       
config_path, checkpoint_path = download_model('faster_rcnn_resnet50_coco', output_dir='models')
from tftrt.examples.object_detection import optimize_model

frozen_graph = optimize_model(
    config_path=config_path,
    checkpoint_path=checkpoint_path,
    use_trt=True,
    precision_mode='FP32',
    calib_images_dir='dataset/calib_dir'
)

from tftrt.examples.object_detection import download_dataset

images_dir, annotation_path = download_dataset('val2014', output_dir='dataset')

from tftrt.examples.object_detection import benchmark_model

statistics = benchmark_model(
    frozen_graph=frozen_graph,
    images_dir=images_dir,
    annotation_path=annotation_path
)

print('avg_latency_ms: ',statistics['avg_latency_ms'])
print('avg_throughput_fps: ',statistics['avg_throughput_fps'])
print('map: ',statistics['map'])

from tftrt.examples.object_detection import test

```
LOG：
```

2019-04-29 09:34:10.987362: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:7 with 10377 MB memory) -> physical GPU (device: 7, name: GeForce GTX 1080 Ti, pci bus id: 0000:8a:00.0, compute capability: 6.1)
  0%|                                                                                                                                                           | 0/4096 [00:00<?, ?it/s]2019-04-29 09:34:15.090991: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node Preprocessor/ResizeToRange/strided_slice_3. Error: Pack node (Preprocessor/ResizeToRange/stack_2) axis attribute is out of bounds: 0
2019-04-29 09:34:15.785398: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:241] Failed to run optimizer ArithmeticOptimizer, stage RemoveStackStridedSliceSameAxis node Preprocessor/ResizeToRange/strided_slice_3. Error: Pack node (Preprocessor/ResizeToRange/stack_2) axis attribute is out of bounds: 0
2019-04-29 09:34:16.334620: I tensorflow/stream_executor/dso_loader.cc:153] successfully opened CUDA library libcublas.so.10 locally
2019-04-29 09:34:16.598452: I tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc:496] Building a new TensorRT engine for TRTEngineOp_0 with batch size 1
2019-04-29 09:34:38.195497: I tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc:496] Building a new TensorRT engine for TRTEngineOp_1 with batch size 100
  2%|██▍                                                                                                                                               | 67/4096 [00:32<05:14, 12.79it/s]Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1334, in _do_call
    return fn(*args)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1319, in _run_fn
    options, feed_dict, fetch_list, target_list, run_metadata)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1407, in _call_tf_sessionrun
    run_metadata)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Incompatible shapes: [17328] vs. [16150]
	 [[{{node Decode/mul_3}}]]
	 [[{{node SecondStagePostprocessor/Reshape_4}}]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 24, in <module>
    annotation_path=annotation_path
  File "/opt/tensorflow/nvidia-examples/tensorrt/tftrt/examples/object_detection/object_detection.py", line 568, in benchmark_model
    feed_dict={tf_input: batch_images})
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 929, in run
    run_metadata_ptr)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1152, in _run
    feed_dict_tensor, options, run_metadata)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1328, in _do_run
    run_metadata)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/client/session.py", line 1348, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Incompatible shapes: [17328] vs. [16150]
	 [[node Decode/mul_3 (defined at /opt/tensorflow/nvidia-examples/tensorrt/tftrt/examples/object_detection/object_detection.py:535) ]]
	 [[node SecondStagePostprocessor/Reshape_4 (defined at /opt/tensorflow/nvidia-examples/tensorrt/tftrt/examples/object_detection/object_detection.py:535) ]]

Caused by op 'Decode/mul_3', defined at:
  File "test.py", line 24, in <module>
    annotation_path=annotation_path
  File "/opt/tensorflow/nvidia-examples/tensorrt/tftrt/examples/object_detection/object_detection.py", line 535, in benchmark_model
    tf.import_graph_def(frozen_graph, name='')
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/importer.py", line 442, in import_graph_def
    _ProcessNewOps(graph)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/importer.py", line 235, in _ProcessNewOps
    for new_op in graph._add_new_tf_operations(compute_devices=False):  # pylint: disable=protected-access
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 3433, in _add_new_tf_operations
    for c_op in c_api_util.new_tf_operations(self)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 3433, in <listcomp>
    for c_op in c_api_util.new_tf_operations(self)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 3325, in _create_op_from_tf_operation
    ret = Operation(c_op, self)
  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 1801, in __init__
    self._traceback = tf_stack.extract_stack()

InvalidArgumentError (see above for traceback): Incompatible shapes: [17328] vs. [16150]
	 [[node Decode/mul_3 (defined at /opt/tensorflow/nvidia-examples/tensorrt/tftrt/examples/object_detection/object_detection.py:535) ]]
	 [[node SecondStagePostprocessor/Reshape_4 (defined at /opt/tensorflow/nvidia-examples/tensorrt/tftrt/examples/object_detection/object_detection.py:535) ]]

```

I noticed that this error did not happen at the very first test image, but after running a few of images (67 images finished). Is it a bug?

**Comment by pooyadavoodi** [comment]
I think this is fixed in TF 1.14. Could you please test again.

This is the tail of the log of my faster_rcnn_resnet50_coco:
```
Loading and preparing results...
DONE (t=0.11s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=9.94s).
Accumulating evaluation results...
DONE (t=1.74s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.265
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.416
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.297
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.075
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.284
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.455
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.238
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.322
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.323
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.090
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.337
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.568
{
    "avg_latency_ms": 29.08458971352608,
    "avg_throughput_fps": 34.38246885548948,
    "map": 0.2651635672111736
}
ASSERTION PASSED: statistics['map'] > (0.264 - 0.005)
PASS faster_rcnn_resnet50_coco_trt_fp16.json
DONE testing faster_rcnn_resnet50_coco_trt_fp16.json
```

The config file:
```
{
  "source_model": {
    "model_name": "faster_rcnn_resnet50_coco",
    "input_dir": "/data/tensorflow/object_detection/models"
  },
  "optimization_config": {
    "use_trt": true,
    "precision_mode": "FP16",
    "override_nms_score_threshold": 0.3,
    "max_batch_size": 1
  },
  "benchmark_config": {
    "images_dir": "/data/coco/coco-2017/coco2017/val2017",
    "annotation_path": "/data/coco/coco-2017/coco2017/annotations/instances_val2017.json",
    "batch_size": 1,
    "image_shape": [640, 640],
    "num_images": 4096,
    "output_path": "stats/faster_rcnn_resnet50_coco_trt_fp16.json"
  },
  "assertions": [
    "statistics['map'] > (0.264 - 0.005)"
  ]
}
```

**Comment by pooyadavoodi** [comment]
Closing. Please reopen in case the issue remains.
